Sources/NDT7Settings.swift

```diff
@@ -113,110 +80,79 @@ public struct NDT7Timeouts {
+}
 
-    /// fqdn
-    public var fqdn: String?
+/// Response type for the new, Locate API V2 response
+public struct NDT7ServerV2: Codable {
+    public var machine: String
+    
+    public var location: NDT7Location
+    
+    public var urls: NDT7LocationUrls
+}
 
-    /// site
-    public var site: String?
+public struct NDT7Location: Codable {
+    public var city: String
+    
+    public var country: String
 }
 
-/// This extension provides helper methods to discover Mlab servers availables.
-extension NDT7Server {
+/// Upload and download URLs for the V2 Location API
+public struct NDT7LocationUrls: Codable {
+    public var downloadUrl: String
+    
+    public var uploadUrl: String
+    
+    public var insecureDownloadUrl: String
+    
+    public var insecureUploadUrl: String
+    
+    enum CodingKeys: String, CodingKey {
+        case downloadUrl = "wss:///ndt/v7/download"
+        case uploadUrl = "wss:///ndt/v7/upload"
+        case insecureDownloadUrl = "ws:///ndt/v7/download"
+        case insecureUploadUrl = "ws:///ndt/v7/upload"
+    }
+}
 
-    /// Discover the closer Mlab server available or using geo location to get a random server from a list of the closer servers.
+extension NDT7ServerV2 {
+    
+    /// Calls the Location V2 API to find the M-Labs recommended servers to call.
+    /// These servers have on them authenticated URLs to call
     /// - parameter session: URLSession object used to request servers, using URLSession.shared object as default session.
-    /// - parameter geoOptions: true to use a list of servers based in geo location, otherwise the function will work trying to get the closer server.
-    /// - parameter retray: true to use a list of servers based in geo location, otherwise the function will work trying to get the closer server.
-    /// - parameter geoOptionsChangeInRetray: Alternate geo options for retray.
-    /// - parameter useNDT7ServerCache: uses the NDT7 Server cached if exist and if the request/s fail.
     /// - parameter completion: callback to get the NDT7Server and error message.
-    /// - parameter server: NDT7Server object representing the Mlab server.
+    /// - parameter server: An array of NDT7ServerV2 object representing the Mlab servers available to call.
     /// - parameter error: if any error happens, this parameter returns the error.
-    public static func discover<T: URLSessionNDT7>(session: T = URLSession.shared as! T,
-                                                   withGeoOptions geoOptions: Bool,
-                                                   retray: UInt = 0,
-                                                   geoOptionsChangeInRetray: Bool = false,
-                                                   useNDT7ServerCache: Bool = false,
-                                                   _ completion: @escaping (_ server: NDT7Server?, _ error: NSError?) -> Void) -> URLSessionTaskNDT7 {
-        let retray = min(retray, 4)
```

Comment:
any reason why the option to retry and cache was removed?

Reply:
Yes, the access tokens may not be long-lived and a cache would be caching a token that might be expired: https://www.measurementlab.net/develop/locate-v2/

Reply:
Our original intent was to retry by trying all 4 servers instead of just the first one returned if the first one fails.

Reply:
The retry is not regarding the download and upload test, it is regarding getting the servers and it was pretty unstable before, so, it's better to maintain this logic to retry to get the servers for NDT7.